src/core/unicode.js

```diff
@@ -1630,8 +1630,34 @@ function reverseIfRtl(chars) {
   return s;
 }
 
+function normalizeUnicode(str, normalizedUnicodesMap) {
```

Comment:
For performance reasons I still think we need a fast-path for the `str.length === 1` case here, i.e. what you had in a prior version of this patch:
```javascript
if (str.length === 1) {
  if (normalizedUnicodesMap[str] !== undefined) {
    return normalizedUnicodesMap[str];
  }
  return str;
}
```